query/physicalplan/aggregate.go

```diff
@@ -315,14 +320,25 @@ func (a *HashAggregate) Callback(ctx context.Context, r arrow.Record) error {
 			}
 		}
 
-		if a.columnToAggregate.MatchColumn(field.Name) {
-			columnToAggregate = r.Column(i)
-			aggregateFieldFound = true
+		for j, col := range a.columns {
+			if a.finalStage {
+				if col.resultName == field.Name {
+					columnToAggregate[j] = r.Column(i)
+					aggregateFieldFound[j] = true
+				}
+			} else {
+				if col.expr.MatchColumn(field.Name) {
+					columnToAggregate[j] = r.Column(i)
+					aggregateFieldFound[j] = true
+				}
+			}
 		}
 	}
 
-	if !aggregateFieldFound {
-		return errors.New("aggregate field not found, aggregations are not possible without it")
+	for _, found := range aggregateFieldFound {
```

Comment:
nit: another option is to use a counter instead of `n` booleans. Avoids allocations as well